File: src/go/fastinvsqrt.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"unsafe"
)

func fastInvSqrt(x float32) float32 {
	i := *(*int32)(unsafe.Pointer(&x))
	i = 0x5f3759df - i>>1
	y := *(*float32)(unsafe.Pointer(&i))
	return y * (1.5 - 0.5*x*y*y)
}

func main() {
	scanner := bufio.NewScanner(os.Stdin)
	for scanner.Scan() {
		if x, err := strconv.ParseFloat(scanner.Text(), 32); err == nil {
			fmt.Println(fastInvSqrt(float32(x)))
		}
	}
}

File: src/nim/fastinvsqrt.nim
import strutils

proc fastInvSqrt(x: float32): float32 =
  let i = cast[int32](x)
  let j = 0x5f3759df - i shr 1
  let y = cast[float32](j)
  return y * (1.5 - 0.5 * x * y * y)

while true:
  try:
    let input = stdin.readLine
    try:
      echo fastInvSqrt(parseFloat(input))
    except ValueError:
      continue
  except IOError:
    break
